### ISSUE TITLE
winpty_agent.exe crashes when running Far Manager and resizing terminal 

### ISSUE THREAD

**Issue by aleherb** [rprichard/winpty#30]
When running Far Manager (http://www.farmanager.com/) under control of winpty, increasing the vertical terminal size (e.g. mintty) crashes winpty_agent.exe due to an assertion failure:

Assertion failed: bufferRect.contains(newWindowRect), file Win32Console.cc, line 222

The reason seems to be that Far Manager changes the console buffer size to the window size, which winpty cannot handle properly.

**Comment by rprichard** [comment]
Yeah, winpty doesn't currently handle programs that change the window/buffer sizes.  I *think* this can be fixed.

**Comment by rprichard** [comment]
I have some experimental changes to winpty that detect when the console buffer is resized and switches to a different mode that makes no effort to track scrolling.  These changes fix the crashing.

Unfortunately, opening Far Manager under winpty in Windows 10 intermittently hangs `conhost.exe`.  (It's the same underlying issue as https://github.com/rprichard/winpty/issues/31).  Strictly speaking, this is a Windows bug, not that it matters to winpty users.  If you have Windows 10, you can see the problem easily:
 - (For practical reasons: Make sure you can begin text selection in the console with a simple left-click-and-drag -- no right-clicking.)
 - In a console, run "C:\Program Files\Far Manager\Far.exe"
 - Immediately, before Far Manager can start, click-and-drag to begin selection.

At this point, the console window becomes non-responsive, and Far Manager does not appear.  Closing the console window isn't possible -- to clean up, conhost.exe must be killed from the Task Manager.

winpty hits this hang because it's using selection to momentarily freeze console output while it scrapes.  It needs to freeze console output so it gets a consistent view of the console buffer content.  It can't read the whole buffer at once, because reads are limited to something like 30KB.  (In older versions of Windows, anyway.  Maybe that's changed in Windows 10...)

**Comment by rprichard** [comment]
The original complaint should be fixed by https://github.com/rprichard/winpty/commit/d4640890cf00346d403c9688a057013b7cf2f3f4.  Resizing the terminal no longer crashes winpty.

Starting and stopping Far Manager can hang conhost.exe in Windows 10, though, so I'll leave this issue open for now.  A workaround is to enable legacy mode.

**Comment by rprichard** [comment]
IIRC, newer releases of Windows 10 fix the `conhost.exe` hang.
